src/core/DataFile.cpp

```diff
@@ -901,6 +901,29 @@ void DataFile::upgrade_1_1_91()
 }
 
 
+void DataFile::upgrade_1_2_0_rc3()
+{
+	// Upgrad from earlier bbtrack beat note behaviour of adding
+	// steps if a note is placed after the last step.
+	QDomNodeList list = elementsByTagName( "bbtrack" );
+	for( int i = 0; !list.item( i ).isNull(); ++i )
+	{
+		list = elementsByTagName( "pattern" );
+		for( int i = 0; !list.item( i ).isNull(); ++i )
+		{
+			int patternLength, steps;
+			QDomElement el = list.item( i ).toElement();
+			for( int i = 0; !list.item( i ).isNull(); ++i )
+			{
+				patternLength = el.attribute( "len" ).toInt();
```

Comment:
Do all `"pattern"` attributes contain an integer value (never blank?).  We just need to make sure we [sanitize our input](https://github.com/zonkmachine/lmms/blob/f2b611b0e8f6848e64d233fbd9c81b509a305268/src/core/DataFile.cpp#L413).  I think this change is fine, but any time we assume a String is an integer we need to make sure we don't segfault on something simple.